cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -143,7 +143,7 @@ controller, and serviceaccounts controller.`,
 			}
 			// add feature enablement metrics
 			utilfeature.DefaultMutableFeatureGate.AddMetrics()
-			return Run(context.Background(), c.Complete())
```

Comment:
Actually I'm not quite buy this because the component is depend on the library, which may lead to unexpected behaviors if the library updates. I don't dig into the details, but I think using a background context doesn't lead to any problems here.  My two cents.

Reply:
Thank you for the review comment. I agree with you.

Reply:
Thanks for your review. I agree with your opinion. But if the upstream component has ctx, we should pass the upstream ctx down, which will be better. I read the source code, this ctx will definitely be initialized, there will be no ctx == nil.

Reply:
I think the point is what is the benefit for being correct vs assuming a risk of depending on a third party library that may change some behavior and break our controller (one that clusters depend completely on them)

Reply:
I discussed with @kerthcet offline. It is more reasonable not to rely on third-party libraries, so I will close this PR. Thank you for your review @aojea @kerthcet